### PR TITLE
Added Task::getUid method

### PR DESCRIPTION
**Pull request by dbarashev**
... and returned back lots of tests :)

### CODE REVIEW

File: ganttproject/src/main/java/net/sourceforge/ganttproject/task/TaskManagerImpl.java [hunk 1]
```diff
@@ -355,15 +355,18 @@ public TaskBuilder newTaskBuilder() {
...
-
         if (myPrototype != null) {
           myId = myPrototype.getTaskID();
         }
         if (myId == null || myTaskMap.getTask(myId) != null) {
           myId = getAndIncrementId();
         }
 
-        TaskImpl task = new GanttTask("", CalendarFactory.createGanttCalendar(), 1, TaskManagerImpl.this, myId);
+        var startDate = myStartDate != null
```
Comment (npostnikova) [inline ganttproject/src/main/java/net/sourceforge/ganttproject/task/TaskManagerImpl.java:365]: Should this part of the code be removed from `TaskManager.TaskBuilder.setupNewTask` then? The method is not used anywhere but here

File: ganttproject-tester/test/biz/ganttproject/core/calendar/WeekendCalendarImplTest.java [hunk 2]
```diff
@@ -56,15 +56,19 @@ public DateFormat getShortDateFormat() {
...
       CalendarEvent.newEvent(CalendarFactory.createGanttCalendar(2014, 1, 14).getTime(), false, CalendarEvent.Type.NEUTRAL, "Feb 14", null),
       CalendarEvent.newEvent(CalendarFactory.createGanttCalendar(2014, 2, 8).getTime(), true, CalendarEvent.Type.HOLIDAY, "Mar 8", null),
       CalendarEvent.newEvent(CalendarFactory.createGanttCalendar(2014, 2, 8).getTime(), false, CalendarEvent.Type.WORKING_DAY, "Mar 8, 2014", null),
       CalendarEvent.newEvent(CalendarFactory.createGanttCalendar(2014, 2, 9).getTime(), false, CalendarEvent.Type.HOLIDAY, "Mar 9, 2014", null),
       CalendarEvent.newEvent(CalendarFactory.createGanttCalendar(2014, 3, 12).getTime(), true, CalendarEvent.Type.WORKING_DAY, "Apr 12", null),
       CalendarEvent.newEvent(CalendarFactory.createGanttCalendar(2014, 3, 12).getTime(), false, CalendarEvent.Type.HOLIDAY, "Apr 12, 2014", null)
-  );
+    );
+  }
+  private static List<CalendarEvent> TEST_EVENTS = createTestEvents();
```
Comment (npostnikova) [inline ganttproject-tester/test/biz/ganttproject/core/calendar/WeekendCalendarImplTest.java:71]: Could you please explain the trick? :)
Why to create `TEST_EVENTS` and use it in the first method only?
Shouldn't `TEST_EVENTS_RECURRING_FIRST` be converted to method as well for consistency?

File: ganttproject-tester/test/biz/ganttproject/core/calendar/WeekendCalendarImplTest.java [hunk 3]
```diff
@@ -40,7 +40,7 @@
  * @author dbarashev (Dmitry Barashev)
  */
 public class WeekendCalendarImplTest extends TestCase {
-  static {
+  protected void setUp() {
```
Comment (npostnikova) [inline ganttproject-tester/test/biz/ganttproject/core/calendar/WeekendCalendarImplTest.java:43]: An explicit override would be nice :)

**Review by npostnikova — APPROVED** [review]
Looks good overall!

File: ganttproject-tester/test/biz/ganttproject/core/calendar/WeekendCalendarImplTest.java (hunk 2 above)
Comment (dbarashev) [reply ganttproject-tester/test/biz/ganttproject/core/calendar/WeekendCalendarImplTest.java]: Thanks, I replaced both static lists with the method calls.
The issue which I struggled with was that these tests were failing if I ran them all (like `gradle test`). However, with a filter (even a very simple `gradle test --tests biz*`) they were passing. 
I didn't get the reasons exactly, but I think that there were side effects of setting the static member `ourLocaleApi` from different tests. That's why I replaced static instances and static setup with instances created per each test.

File: ganttproject/src/main/java/net/sourceforge/ganttproject/task/TaskManagerImpl.java (hunk 1 above)
Comment (dbarashev) [reply ganttproject/src/main/java/net/sourceforge/ganttproject/task/TaskManagerImpl.java:365]: done, thanks

File: ganttproject-tester/test/biz/ganttproject/core/calendar/WeekendCalendarImplTest.java (hunk 3 above)
Comment (dbarashev) [reply ganttproject-tester/test/biz/ganttproject/core/calendar/WeekendCalendarImplTest.java:42]: yup

**Pull request merged**